### PR TITLE
Adds the beta group in the available design groups enumeration

### PR DESCRIPTION
**Pull request by antonis**
### Description
Adds the beta group in the available design groups enumeration

### Testing Details
Follow the steps on https://github.com/wordpress-mobile/WordPress-iOS/pull/18683

- [x] Please check here if your pull request includes additional test coverage.
- [x] I have considered updating the `version` in the `.podspec` file.

### CODE REVIEW

File: WordPressKit/SiteDesignServiceRemote.swift [hunk 1]
```diff
@@ -4,6 +4,7 @@ import WordPressShared
 public struct SiteDesignRequest {
     public enum TemplateGroup: String {
         case stable
+        case beta
         case singlePage = "single-page"
```
Comment (antonis) [inline WordPressKit/SiteDesignServiceRemote.swift:8]: Depending on [the outcome of this conversation](https://github.com/wordpress-mobile/WordPress-iOS/pull/18683#discussion_r877831031) we may remove the `singe-page` group

**Review by ovitrif — APPROVED** [review]
Works as expected in the iOS app.
Thank you @antonis !

**Comment by ovitrif** [comment]
Before merging, we need to update the podspec version here: [`WordPressKit.podspec#L5`](https://github.com/wordpress-mobile/WordPressKit-iOS/blob/trunk/WordPressKit.podspec#L5).

**Review by twstokes — APPROVED** [review]
LGTM @antonis! 🎉

**Comment by twstokes** [comment]
@antonis one small thought about versioning: Should this bump the Podspec to `4.53.0`? I see that `4.52.0` has been released but the [release branch](https://github.com/wordpress-mobile/WordPressKit-iOS/tree/release/4.52.0) hasn't been merged back into `trunk` yet.

**Comment by antonis** [comment]
Thank you both for testing and your feedback 🙇 
I'll wait a bit [to resolve this conversation](https://github.com/wordpress-mobile/WordPressKit-iOS/pull/508#discussion_r877836425) and update to the latest Podspec version before merging 🙏

**Pull request merged**